server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -1114,8 +1144,34 @@ void restoreShouldNotAppendMessageToMailboxWhenASingleCriterionDoesntMatch() thr
     class ValidationTest {
 
         @Test
-        void restoreShouldReturnInvalidWhenUserIsInvalid() {
+        void restoreShouldReturnInvalidWhenActionIsMissing() {
             when()
+                .post(USER.asString())
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("action cannot be null or empty"));
+        }
+
+        @Test
+        void restoreShouldReturnInvalidWhenActionIsInValid() {
+            given()
+                .queryParam("action", "invalid action")
+            .when()
+                .post(USER.asString())
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .body("statusCode", is(400))
+                .body("type", is(ErrorResponder.ErrorType.INVALID_ARGUMENT.getType()))
+                .body("message", is("'invalid action' is not a valid action(restore)"));
```

Comment:
A nicer message would be `'invalid action' is not a valid action. Supported values are: (restore)`